webbeans-junit5/src/main/java/org/apache/openwebbeans/junit5/internal/CdiExtension.java

```diff
@@ -104,40 +112,42 @@ public void beforeAll(final ExtensionContext extensionContext)
                 .peek(Supplier::get)
                 .filter(Objects::nonNull)
                 .toArray(Closeable[]::new);
+        SeContainer container = initializer.initialize();
         if (reusable)
         {
-            reusableContainer = initializer.initialize();
+            reusableContainer = container;
             Runtime.getRuntime().addShutdownHook(new Thread(
                 () -> doClose(reusableContainer), getClass().getName() + "-shutdown"));
         }
         else
         {
-            container = initializer.initialize();
+            testInstanceContainer = container;
         }
     }
 
     @Override
     public void afterAll(final ExtensionContext extensionContext)
     {
-        if (container != null)
+        if (testInstanceContainer != null)
         {
-            doClose(container);
-            container = null;
+            doClose(testInstanceContainer);
+            testInstanceContainer = null;
         }
     }
 
     @Override
     public void beforeEach(final ExtensionContext extensionContext)
     {
-        if (container == null && reusableContainer == null)
+        final SeContainer container = getContainer();
+        if (container == null)
```

Comment:
As you mentioned error handling, it occurred to me, that this `if (container == null)` looks suspicious to me. Would you mind removing it including all duplicates I copied or what is the reason it is here?

I reason that for only the first trigger/callback which is `beforeAll` the container could have to be initialized because it will be called always before any other trigger/callback in the extension. If any other trigger/callback is called, a container must have been initialized before, and therefore cannot be null at this time.

Reply:
sadly it is not the case, CdiExtension can be put on a method, directly or indirectly (through another extension annotation) so we must support the case before/afterAll callbacks are bypassed

Reply:
That sounds to me more like the `beforeAll` callback had missed to find the `@Cdi` annontation(s) by not looking at the right execution context(s in various places up the nesting or inheritance or annotation hierarchies) rather than having not been invoked at all, which in turn sounds to me more like an existing bug or incompleteness. It answers my question whether to remove it or not so let's not change it for now.